File: ios/App/App/TrackModels.swift
import Foundation

struct TrackTimelineEvent: Identifiable, Sendable {
    let id = UUID()
    let status: String
    let at: String
    let message: String?
}

struct TrackSnapshot: Sendable {
    let orderId: String
    let orderNumber: String
    let status: String
    let timeline: [TrackTimelineEvent]
    let etaMin: Int?
    let etaMax: Int?
    let restaurantName: String?
}

File: android/app/src/main/java/com/bhojanos/customer/track/TrackModels.kt
package com.bhojanos.customer.track

data class TrackTimelineEvent(
    val status: String,
    val at: String,
    val message: String?,
)

data class TrackSnapshot(
    val orderId: String,
    val orderNumber: String,
    val status: String,
    val timeline: List<TrackTimelineEvent>,
    val etaMin: Int?,
    val etaMax: Int?,
    val restaurantName: String?,
)
